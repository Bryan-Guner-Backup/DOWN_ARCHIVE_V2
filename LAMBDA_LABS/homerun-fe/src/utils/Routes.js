import React from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';

// Sign up component with form for email and gmail button
import SignUp from '../components/auth/SignUp';
import SignUpPartTwo from '../components/auth/SignUpPartTwo';
import Dashboard from '../components/dashboard/Dashboard.js';

// Sign in landing page with two buttons email/gmail
import SignInLanding from '../components/auth/SignIn-Landing';

// Home component is where the beginning of the app starts
import Home from '../components/marketing/Home.js';

// Household component
import Household from '../components/household/Household';

// Header
import Header from '../components/header/Header.js';

//Error 404 Page
import NotFound from '../components/dashboard/NotFound.js';

//About us page from the marketing side
import AboutUs from '../components/marketing/AboutUs.js';

import ForgotPW from '../components/auth/Forgot-Password.js';
import ResetPW from '../components/auth/Reset-Password.js';
import Auth from '../components/auth/Auth.js';

import ContactUsForm from '../components/marketing/ContactUsForm';

import InviteConfirm from '../components/household/InviteConfirm.js';

const PrivateRoute = ({ component: Component, ...rest }) => (
  <Route
    {...rest}
    render={(props) =>
      localStorage.getItem('token') ? (
        <Component {...props} />
      ) : (
        <Redirect to="/" />
      )
    }
  />
);

// Header component needs to render on multiple routes and since the Switch already has a dashboard route and household route
// the header component needs to render outside the switch
const Routes = () => {
  return (
    <>
      <PrivateRoute
        path={['/dashboard', '/household']}
        component={Header}
      />
      <Switch>
        <Route exact path="/" component={Home} />
        <Route exact path="/signin" component={SignInLanding} />
        <Route path="/signup" component={SignUp} />
        <Route path="/error" component={NotFound} />
        <Route path="/contact" component={ContactUsForm} />
        <Route path="/forgot-password" component={ForgotPW} />
        <Route path="/reset-password/:hash" component={ResetPW} />
        <Route path="/auth" component={Auth} />
        <Route path="/about" component={AboutUs} />
        <Route path="/invite/:hash" component={InviteConfirm} />
        <Route path="/confirm/:hash" component={SignUpPartTwo} />
        <PrivateRoute path="/dashboard" component={Dashboard} />
        <PrivateRoute path="/household" component={Household} />
      </Switch>
    </>
  );
};

export default Routes;
