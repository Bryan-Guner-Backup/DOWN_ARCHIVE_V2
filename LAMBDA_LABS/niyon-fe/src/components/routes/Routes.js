import React from 'react'
import Marketing from '../marketing/Marketing'
import About from '../about/About'
import Home from '../home/Home'
import Profile from '../profile/Profile'
import Search from '../search/Search'
import News from '../news/News'
import { Route, Switch } from 'react-router-dom'
import { PrivateRoute } from '../privateRoute/PrivateRoute'
import Registration from '../registration/Registration'
import Login from '../login/Login'
import ChatRoom from '../chat/ChatRoom'
import RoomSelect from '../chat/RoomSelect'

function Routes (props) {
  const token = window.localStorage.getItem('token')
  const authorized = !!token
  return (
    <Switch>
      <Route exact path="/" component={Marketing} />
      <Route path="/about" component={About} />
      <Route path="/registration" component={Registration} />
      <Route path="/login" component={Login} />
      <PrivateRoute
        path="/home"
        isAuthenticated={authorized}
        component={Home}
      />
      <PrivateRoute
        path="/profile"
        isAuthenticated={authorized}
        component={Profile}
      />
      <PrivateRoute
        path="/search"
        isAuthenticated={authorized}
        component={Search}
      />
      <PrivateRoute
        path="/news"
        isAuthenticated={authorized}
        component={News}
      />
      <PrivateRoute
        path="/roomselect"
        isAuthenticated={authorized}
        component={RoomSelect}
      />
      <PrivateRoute
        path="/chatroom/:id/:roomName/:roomId"
        isAuthenticated={authorized}
        component={ChatRoom}
      />
    </Switch>
  )
}

export default Routes
