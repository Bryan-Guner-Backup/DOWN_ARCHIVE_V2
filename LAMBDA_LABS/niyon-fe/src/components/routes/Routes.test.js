import React from 'react'
import toJSON from 'enzyme-to-json'
import { shallow } from 'enzyme'
import Routes from './Routes'

describe('<Routes />', () => {
  let component
  beforeEach(() => {
    component = shallow(<Routes />)
  })

  it('should pass snapshot testing', () => {
    expect(toJSON(component)).toMatchSnapshot()
  })

});
