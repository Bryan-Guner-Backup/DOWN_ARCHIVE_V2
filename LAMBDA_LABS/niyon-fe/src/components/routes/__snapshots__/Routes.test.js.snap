// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes /> should pass snapshot testing 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    path="/about"
  />
  <Route
    component={[Function]}
    path="/registration"
  />
  <Route
    component={[Function]}
    path="/login"
  />
  <PrivateRoute
    component={[Function]}
    isAuthenticated={false}
    path="/home"
  />
  <PrivateRoute
    component={[Function]}
    isAuthenticated={false}
    path="/profile"
  />
  <PrivateRoute
    component={[Function]}
    isAuthenticated={false}
    path="/search"
  />
  <PrivateRoute
    component={[Function]}
    isAuthenticated={false}
    path="/news"
  />
  <PrivateRoute
    component={[Function]}
    isAuthenticated={false}
    path="/roomselect"
  />
  <PrivateRoute
    component={[Function]}
    isAuthenticated={false}
    path="/chatroom/:id/:roomName/:roomId"
  />
</Switch>
`;
